### ISSUE TITLE
resuming the harvest

### ISSUE THREAD

**Issue by igor-kamil** [caseyamcl/phpoaipmh#23]
Hi!
Did anyone of you try to implement resuming the harvest process after fail (f.e. connection lost)?
We are now harvesting big set - which is failing on "cURL error 56" (Failure with receiving network data.)
It would be great, if it would be possible to initialize the harvest just with resumption token.
Or any other idea?

**Comment by caseyamcl** [comment]
Hi @igor-kamil -- that is a good idea.  Let me look into this weekend and get back to you.  At first glance, I think this would be pretty easy to implement.

**Comment by igor-kamil** [comment]
I'm looking forward! Thank you @caseyamcl

**Comment by caseyamcl** [comment]
Well, @igor-kamil .. first glances can be misleading :).  Let me make sure I understand the issue and possible remedy.

Are you using Guzzle or cURL to do the request?  I'm guessing cURL since you're getting a raw cURL error.  But, I need to make sure.

What I believe is happening (and you can verify) is that the client is timing out before receiving all of the data per request.  This would happen if each page of data from your OAI-PMH provider is large.  If this is the case, then upping the timeout for the cURL adapter would probably work.  It is hard-coded to 60, but I can make it configurable.

If you are using Guzzle, this is easier to fix.  Also, I can provide some assistance with retrying requests (possible for the Guzzle client; not so easy with cURL).

**Comment by caseyamcl** [comment]
So, a couple of things.

1. I just released v2.4 with ability to set custom request parameters.  Refer to the *Customizing Default Request Parameters* section in the [README.md](https://github.com/caseyamcl/phpoaipmh/blob/master/README.md) file for details.  I'd suggest trying to increase the request timeout from the default of 60 seconds and see if that makes a difference.

2. If that doesn't work, there is a way to retry requests if you are using Guzzle.  You can use the following code to recover from errors and attempt the last request again.  I'll work on making a simpler API for this in v3, or you are welcome to submit a PR if you want.  In the meantime, this *should* work:

First, make sure you've installed the Guzzle Retry-Subscriber in your `composer.json`:

```json
require: {
   "guzzlehttp/retry-subscriber": "~2.0"
}
```

Then, you can do something like this:

```php

// Create a Retry Guzzle Subscriber that retries the last request on any failure event
$retrySubscriber = new \GuzzleHttp\Subscriber\Retry\RetrySubscriber([
    'delay' => function($numRetries, \GuzzleHttp\Event\AbstractTransferEvent $event) {
        $waitSecs = $event->getResponse()->getHeader('Retry-After') ?: '5'; // default try again in 5secs
        return ($waitSecs * 1000) + 1000; // wait one second longer than the server said to
    },
    'filter' => \GuzzleHttp\Subscriber\Retry\RetrySubscriber::createIdempotentFilter(),
]);

// Manually create a Guzzle HTTP adapter
$guzzleAdapter = new \Phpoaipmh\HttpAdapter\Guzzle();
$guzzleAdapter->getGuzzleClient()->getEmitter()->attach($retrySubscriber);

// Build PHPOAIPMH client and endpoint
$client  = new \Phpoaipmh\Client('http://some.service.com/oai', $guzzleAdapter);
$endpoint = new \Phpoaipmh\Endpoint($client);
```

**Comment by caseyamcl** [comment]
HI @igor-kamil  - I'm going to close this issue, since I haven't heard back from you.  If you still haven't resolved this issue, let me know, and we can re-open it.  Thanks.
